src/requirements/Uniswap/hooks/useParseVaultAddress.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+
+// Note: We might want to put these kind of constants to `@guildxyz/types/consts`
+export const ADDRESS_REGEX = /0x[0-9a-f]{40}/i
+
+export function useParseVaultAddress(baseFieldPath: string) {
+  const { setValue } = useFormContext()
+  const lpVaultInputValue = useWatch({ name: `${baseFieldPath}.data.lpVault` })
+
+  const lpVaultAddress = useMemo(() => {
```

Comment:
This is a bit scary in my opinion, not sure if it's a good practice to change an input field's value inside a useMemo 👀 Couldn't we just move this logic to a useEffect & watch the `data.lpVault` input's value with a useWatch?

Reply:
> I'm not even sure if we need to change the input value at all

Yeah, let's don't change it then, and that way we can keep this useMemo 👌

Reply:
Agreed! Removed in https://github.com/guildxyz/guild.xyz/pull/1158/commits/13732968760b8ae55283c676a59ddf0379f376ea 👍 
Also made a small copy change from "LP Vault" to "Pool address or URL"